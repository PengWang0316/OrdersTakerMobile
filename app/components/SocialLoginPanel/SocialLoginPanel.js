import React, { Fragment } from 'react';
import { View, Text } from 'react-native';
import { Button, Icon } from 'react-native-elements';

import styles from './Styles';
import Theme from '../../Theme';

export const SocialLoginPanel = () => (
  <Fragment>
    <View>
      <Text style={styles.title}>Social Login</Text>
    </View>
    <View>
      <Button
        buttonStyle={[styles.loginButton, styles.facebookBtn]}
        title="Sign in with Facebook"
        icon={(
          <Icon
            type="font-awesome"
            name="facebook"
            color={Theme.primary.contrastText}
          />
        )}
      />
    </View>

    <View>
      <Button
        buttonStyle={[styles.loginButton, styles.googleBtn]}
        title="Sign in with Google"
        icon={(
          <Icon
            type="entypo"
            name="google-"
            color={Theme.primary.contrastText}
          />
        )}
      />
    </View>
  </Fragment>
);
export default SocialLoginPanel;
