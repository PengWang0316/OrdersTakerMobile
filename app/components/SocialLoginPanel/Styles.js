import { StyleSheet } from 'react-native';
import Theme from '../../Theme';

export default StyleSheet.create({
  title: {
    fontWeight: 'bold',
    fontSize: 20,
    marginBottom: 15,
    color: Theme.primary.main,
  },
  loginButton: {
    marginBottom: 15,
    borderRadius: 15,
  },
  facebookBtn: {
    backgroundColor: '#3B5998',
  },
  googleBtn: {
    backgroundColor: '#dd4b39',
  },
});
