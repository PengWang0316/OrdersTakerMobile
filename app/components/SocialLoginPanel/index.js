import SocialLoginPanel from './SocialLoginPanel';

export default SocialLoginPanel;
