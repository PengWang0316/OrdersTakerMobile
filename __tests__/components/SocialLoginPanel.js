import React from 'react';
import renderer from 'react-test-renderer';

import { SocialLoginPanel } from '../../app/components/SocialLoginPanel/SocialLoginPanel';

jest.mock('Text', () => 'Text');
jest.mock('View', () => 'View');
// jest.mock('react-native-elements', { Icon: () => 'Icon', Button: () => 'Button' });

describe('SocialLoginPanel Test', () => {
  test('Snapshot', () => expect(renderer.create(<SocialLoginPanel />).toJSON()).toMatchSnapshot());
});
