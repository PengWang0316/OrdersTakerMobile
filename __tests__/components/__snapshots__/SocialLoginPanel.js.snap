// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginPanel Test Snapshot 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "color": "#6d9cb5",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 15,
        }
      }
    >
      Social Login
    </Text>
  </View>,
  <View>
    <View
      style={
        Array [
          undefined,
          false,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderRadius": 3,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "borderRadius": 15,
                  "marginBottom": 15,
                },
                Object {
                  "backgroundColor": "#3B5998",
                },
              ],
              false,
              false,
              false,
              undefined,
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                ]
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                undefined,
                false,
                false,
              ]
            }
          >
            Sign in with Facebook
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View>
    <View
      style={
        Array [
          undefined,
          false,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderRadius": 3,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "borderRadius": 15,
                  "marginBottom": 15,
                },
                Object {
                  "backgroundColor": "#dd4b39",
                },
              ],
              false,
              false,
              false,
              undefined,
            ]
          }
        >
          <View>
            <View
              style={
                Array [
                  false,
                  false,
                  false,
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  },
                ]
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ],
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                undefined,
                false,
                false,
              ]
            }
          >
            Sign in with Google
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
